### PR TITLE
Remove redundant server constructor type

### PR DESCRIPTION
**Pull request by always-maap**
- Remove "NextServerConstructor" type which the "dev" property already existed in "ServerConstructor" and fix its use cases.
- Replace "any" with "ServerConstructor" type in start-server.

Thanks

### CODE REVIEW

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| buildDuration | 12.4s | 12.4s | -2ms |
| buildDurationCached | 3.1s | 3.1s | -3ms |
| nodeModulesSize | 47 MB | 47 MB | -53 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.619 | 2.607 | -0.01 |
| / avg req/sec | 954.45 | 959.12 | +4.67 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.289 | 1.306 | ⚠️ +0.02 |
| /error-in-render avg req/sec | 1940.01 | 1914.14 | ⚠️ -25.87 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| 745.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 26.6 kB | 26.6 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 70.4 kB | 70.4 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| polyfills-a4..dd70.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| _app-HASH.js gzip | 979 B | 979 B | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 312 B | 312 B | ✓ |
| css-HASH.js gzip | 329 B | 329 B | ✓ |
| dynamic-HASH.js gzip | 2.67 kB | 2.67 kB | ✓ |
| head-HASH.js gzip | 351 B | 351 B | ✓ |
| hooks-HASH.js gzip | 918 B | 918 B | ✓ |
| image-HASH.js gzip | 4.14 kB | 4.14 kB | ✓ |
| index-HASH.js gzip | 261 B | 261 B | ✓ |
| link-HASH.js gzip | 1.66 kB | 1.66 kB | ✓ |
| routerDirect..HASH.js gzip | 318 B | 318 B | ✓ |
| script-HASH.js gzip | 387 B | 387 B | ✓ |
| withRouter-HASH.js gzip | 320 B | 320 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 13 kB | 13 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| _buildManifest.js gzip | 492 B | 492 B | ✓ |
| Overall change | 492 B | 492 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| index.html gzip | 541 B | 541 B | ✓ |
| link.html gzip | 552 B | 552 B | ✓ |
| withRouter.html gzip | 533 B | 533 B | ✓ |
| Overall change | 1.63 kB | 1.63 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Webpack 4 Mode</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| buildDuration | 10.1s | 10s | -28ms |
| buildDurationCached | 4.1s | 4.1s | -64ms |
| nodeModulesSize | 47 MB | 47 MB | -53 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 2.66 | 2.639 | -0.02 |
| / avg req/sec | 939.96 | 947.43 | +7.47 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 1.287 | 1.31 | ⚠️ +0.02 |
| /error-in-render avg req/sec | 1942.83 | 1908.53 | ⚠️ -34.3 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| 16.HASH.js gzip | 186 B | 186 B | ✓ |
| 677f882d2ed8..HASH.js gzip | 14.1 kB | 14.1 kB | ✓ |
| framework.HASH.js gzip | 41.9 kB | 41.9 kB | ✓ |
| main-HASH.js gzip | 13.8 kB | 13.8 kB | ✓ |
| webpack-HASH.js gzip | 1.19 kB | 1.19 kB | ✓ |
| Overall change | 71.2 kB | 71.2 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| polyfills-a4..dd70.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| _app-HASH.js gzip | 964 B | 964 B | ✓ |
| _error-HASH.js gzip | 3.8 kB | 3.8 kB | ✓ |
| amp-HASH.js gzip | 552 B | 552 B | ✓ |
| css-HASH.js gzip | 333 B | 333 B | ✓ |
| dynamic-HASH.js gzip | 2.87 kB | 2.87 kB | ✓ |
| head-HASH.js gzip | 3.06 kB | 3.06 kB | ✓ |
| hooks-HASH.js gzip | 924 B | 924 B | ✓ |
| index-HASH.js gzip | 231 B | 231 B | ✓ |
| link-HASH.js gzip | 1.64 kB | 1.64 kB | ✓ |
| routerDirect..HASH.js gzip | 298 B | 298 B | ✓ |
| script-HASH.js gzip | 3.03 kB | 3.03 kB | ✓ |
| withRouter-HASH.js gzip | 295 B | 295 B | ✓ |
| 30809af5c834..565.css gzip | 125 B | 125 B | ✓ |
| Overall change | 18.1 kB | 18.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| _buildManifest.js gzip | 500 B | 500 B | ✓ |
| Overall change | 500 B | 500 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| index.html gzip | 584 B | 584 B | ✓ |
| link.html gzip | 597 B | 597 B | ✓ |
| withRouter.html gzip | 578 B | 578 B | ✓ |
| Overall change | 1.76 kB | 1.76 kB | ✓ |

</details>

</details>
Commit: 04091c35856701c3bb32104f04d69452d15aa9c8

File: packages/next/server/next.ts [hunk 1]
```diff
@@ -103,7 +96,7 @@ export class NextServer {
   }
 
   private async createServer(
-    options: NextServerConstructor & {
+    options: ServerConstructor & {
       conf: NextConfig
       isNextDevCommand?: boolean
```
[the remark this answers is not in the document]
Comment (always-maap) [reply packages/next/server/next.ts:101]: I'm still reading the code base but it's been used in [next-dev-server](https://github.com/vercel/next.js/blob/canary/packages/next/server/dev/next-dev-server.ts#L87) as far as I know.

**Review by styfle — APPROVED** [review]

**Comment by ijjk** [comment]
# Stats from current PR

<details>
<summary><strong>Default Build</strong> (Decrease detected ✓)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| buildDuration | 16.3s | 16.5s | ⚠️ +141ms |
| buildDurationCached | 4.1s | 4s | -131ms |
| nodeModulesSize | 182 MB | 182 MB | -53 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall decrease ⚠️</summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.7 | 4.014 | ⚠️ +0.31 |
| / avg req/sec | 675.7 | 622.83 | ⚠️ -52.87 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.135 | 2.174 | ⚠️ +0.04 |
| /error-in-render avg req/sec | 1170.75 | 1150.09 | ⚠️ -20.66 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| 779.HASH.js gzip | 179 B | 179 B | ✓ |
| framework-HASH.js gzip | 42.2 kB | 42.2 kB | ✓ |
| main-HASH.js gzip | 26.8 kB | 26.8 kB | ✓ |
| webpack-HASH.js gzip | 1.45 kB | 1.45 kB | ✓ |
| Overall change | 70.6 kB | 70.6 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| polyfills-a4..dd70.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| _app-HASH.js gzip | 977 B | 977 B | ✓ |
| _error-HASH.js gzip | 194 B | 194 B | ✓ |
| amp-HASH.js gzip | 311 B | 311 B | ✓ |
| css-HASH.js gzip | 328 B | 328 B | ✓ |
| dynamic-HASH.js gzip | 2.67 kB | 2.67 kB | ✓ |
| head-HASH.js gzip | 351 B | 351 B | ✓ |
| hooks-HASH.js gzip | 918 B | 918 B | ✓ |
| image-HASH.js gzip | 4.14 kB | 4.14 kB | ✓ |
| index-HASH.js gzip | 260 B | 260 B | ✓ |
| link-HASH.js gzip | 1.66 kB | 1.66 kB | ✓ |
| routerDirect..HASH.js gzip | 320 B | 320 B | ✓ |
| script-HASH.js gzip | 386 B | 386 B | ✓ |
| withRouter-HASH.js gzip | 319 B | 319 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 13 kB | 13 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| _buildManifest.js gzip | 493 B | 493 B | ✓ |
| Overall change | 493 B | 493 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| index.html gzip | 538 B | 538 B | ✓ |
| link.html gzip | 550 B | 550 B | ✓ |
| withRouter.html gzip | 532 B | 532 B | ✓ |
| Overall change | 1.62 kB | 1.62 kB | ✓ |

</details>

</details>
<hr/>
<details>
<summary><strong>Default Build with SWC</strong> (Increase detected ⚠️)</summary>

<br/>

<details>
<summary><strong>General</strong> Overall decrease ✓</summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| buildDuration | 8.3s | 8.6s | ⚠️ +266ms |
| buildDurationCached | 4s | 4.2s | ⚠️ +129ms |
| nodeModulesSize | 182 MB | 182 MB | -53 B |

</details>

<details>
<summary><strong>Page Load Tests</strong> Overall increase ✓</summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| / failed reqs | 0 | 0 | ✓ |
| / total time (seconds) | 3.724 | 3.607 | -0.12 |
| / avg req/sec | 671.3 | 693.18 | +21.88 |
| /error-in-render failed reqs | 0 | 0 | ✓ |
| /error-in-render total time (seconds) | 2.056 | 2.043 | -0.01 |
| /error-in-render avg req/sec | 1215.74 | 1223.77 | +8.03 |

</details>

<details>
<summary><strong>Client Bundles (main, webpack, commons)</strong></summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| 675-HASH.js gzip | 13.8 kB | 13.8 kB | ✓ |
| 770.HASH.js gzip | 178 B | 178 B | ✓ |
| framework-HASH.js gzip | 50.7 kB | 50.7 kB | ✓ |
| main-HASH.js gzip | 34.7 kB | 34.7 kB | ✓ |
| webpack-HASH.js gzip | 1.65 kB | 1.65 kB | ✓ |
| Overall change | 101 kB | 101 kB | ✓ |

</details>

<details>
<summary><strong>Legacy Client Bundles (polyfills)</strong></summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| polyfills-a4..dd70.js gzip | 31 kB | 31 kB | ✓ |
| Overall change | 31 kB | 31 kB | ✓ |

</details>

<details>
<summary><strong>Client Pages</strong></summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| _app-HASH.js gzip | 1.32 kB | 1.32 kB | ✓ |
| _error-HASH.js gzip | 180 B | 180 B | ✓ |
| amp-HASH.js gzip | 315 B | 315 B | ✓ |
| css-HASH.js gzip | 331 B | 331 B | ✓ |
| dynamic-HASH.js gzip | 2.8 kB | 2.8 kB | ✓ |
| head-HASH.js gzip | 356 B | 356 B | ✓ |
| hooks-HASH.js gzip | 637 B | 637 B | ✓ |
| image-HASH.js gzip | 573 B | 573 B | ✓ |
| index-HASH.js gzip | 262 B | 262 B | ✓ |
| link-HASH.js gzip | 2.2 kB | 2.2 kB | ✓ |
| routerDirect..HASH.js gzip | 326 B | 326 B | ✓ |
| script-HASH.js gzip | 393 B | 393 B | ✓ |
| withRouter-HASH.js gzip | 322 B | 322 B | ✓ |
| bb14e60e810b..30f.css gzip | 125 B | 125 B | ✓ |
| Overall change | 10.1 kB | 10.1 kB | ✓ |

</details>

<details>
<summary><strong>Client Build Manifests</strong></summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| _buildManifest.js gzip | 511 B | 511 B | ✓ |
| Overall change | 511 B | 511 B | ✓ |

</details>

<details>
<summary><strong>Rendered Page Sizes</strong></summary>

|  | vercel/next.js canary  | always-maap/next.js canary | Change |
| - | - | - | - |
| index.html gzip | 538 B | 538 B | ✓ |
| link.html gzip | 552 B | 552 B | ✓ |
| withRouter.html gzip | 532 B | 532 B | ✓ |
| Overall change | 1.62 kB | 1.62 kB | ✓ |

</details>

</details>
Commit: ad455937a67a27534a876d3fd0b17c1d6f37ff42

**Review by ijjk — COMMENTED** [review]
Hi, this has been updated in separate PRs so I think we can close this now, thanks for the PR though!

**Pull request closed without merging**